src/Resources/contao/dca/tl_calendar_events.php

```diff
@@ -670,7 +670,18 @@ public function generateAlias($varValue, DataContainer $dc)
 		if ($varValue == '')
 		{
 			$autoAlias = true;
-			$varValue = StringUtil::generateAlias($dc->activeRecord->title);
+			$slugOptions = [];
+			$objCalendar = CalendarModel::findByPk($dc->activeRecord->pid);
+			$objPage = PageModel::findWithDetails($objCalendar->jumpTo);
+			if ($objPage)
+			{
+				$slugOptions['locale'] = $objPage->language;
+				if ($objPage->validAliasCharacters)
+				{
+					$slugOptions['validChars'] = $objPage->validAliasCharacters;
+				}
+			}
+			$varValue = System::getContainer()->get('contao.slug.generator')->generate(StringUtil::stripInsertTags($dc->activeRecord->title), $slugOptions);
```

Comment:
Why did you add `StringUtil::stripInsertTags()` here?

Reply:
Probably because `StringUtil::generateAlias()` also stripped the insert tags in the past.

Reply:
You are right. But `StringUtil::generateAlias()` also did `decodeEntities()` and `restoreBasicEntities()`. What about those?

Reply:
That is not necessary I think because `$dc->activeRecord->title` is already decoded and basic-entity-restored. In my test the title `foo < bar [nbsp] baz {{br}}` correctly generated the alias `foo-bar-baz`